src/shared/components/lollipopMutationPlot/LollipopMutationPlot.tsx

```diff
@@ -80,27 +79,59 @@ export default class LollipopMutationPlot extends React.Component<ILollipopMutat
         }))
     }, {});
 
-    private lollipopLabel(mutationsAtPosition:Mutation[]):string {
-        let proteinChanges = _.uniq(mutationsAtPosition.map(m=>m.proteinChange));
-        proteinChanges.sort();
+    private lollipopLabelText(mutationsAtPosition:Mutation[], size?: number): string {
+        const mutationCountsByProteinChange = countMutationsByProteinChange(mutationsAtPosition);
+
+        // only pick specified number of protein change values
+        const proteinChanges = mutationCountsByProteinChange.map(m => m.proteinChange).slice(0, size);
+
+        // sort alphabetically (to make it easier to find longest common starting substring)
+        const proteinChangesSorted = proteinChanges.slice(0).sort();
 
         let startStr = "";
-        if (proteinChanges.length > 1) {
+        if (proteinChangesSorted.length > 1) {
             // only need to compare first and last element of sorted string list to find longest common starting substring of all of them
             startStr = longestCommonStartingSubstring(
-                proteinChanges[0], proteinChanges[proteinChanges.length - 1]
+                proteinChangesSorted[0], proteinChangesSorted[proteinChangesSorted.length - 1]
             );
         }
-        proteinChanges = proteinChanges.map((s:string)=>s.substring(startStr.length));
 
-        return startStr + proteinChanges.join("/");
+        // remove longest common starting substring from all protein change values
+        const proteinChangesTrimmed = proteinChanges.map((s:string) => s.substring(startStr.length));
+
+        // construct label (sorted by protein change count, not alphabetically)
+        let label = startStr + proteinChangesTrimmed.join("/");
+
+        if (proteinChanges.length < mutationCountsByProteinChange.length) {
+            label = `${label} and ${mutationCountsByProteinChange.length - proteinChanges.length} more`;
+        }
+
+        return label;
+    }
+
+    private lollipopLabelTextAnchor(labelText: string, codon: number, fontSize: number): string {
+        let anchor = "middle";
+        const approxLabelWidth = labelText.length * fontSize;
```

Comment:
For simplicity using an approximate width, instead of calculating the actual (computed) width after rendering.